crates/store/re_datafusion/src/dataframe_query_provider.rs

```diff
@@ -127,15 +137,25 @@ impl Stream for DataframePartitionStream {
             )));
         }
 
-        this.store_output_channel
+        let result = this
+            .store_output_channel
             .poll_recv(cx)
-            .map(|result| Ok(result).transpose())
+            .map(|result| Ok(result).transpose());
+
+        if let Poll::Ready(None) = &result {
+            this_outer.inner = None;
```

Comment:
maybe that's overzealous, but imho it's worth pointing out in the comments that this is the core of the mem-leak-reduction workaround. Almost missed it and was wondering where the fix is